src/frontend/src/handler/create_sql_function.rs

```diff
@@ -0,0 +1,185 @@
+        Some(CreateFunctionReturns::Value(data_type)) => {
+            return_type = bind_data_type(&data_type)?;
+            Kind::Scalar(ScalarFunction {})
+        }
+        Some(CreateFunctionReturns::Table(columns)) => {
+            if columns.len() == 1 {
+                // return type is the original type for single column
+                return_type = bind_data_type(&columns[0].data_type)?;
+            } else {
+                // return type is a struct for multiple columns
+                let datatypes = columns
+                    .iter()
+                    .map(|c| bind_data_type(&c.data_type))
+                    .collect::<Result<Vec<_>>>()?;
+                let names = columns
+                    .iter()
+                    .map(|c| c.name.real_value())
+                    .collect::<Vec<_>>();
+                return_type = DataType::new_struct(datatypes, names);
+            }
+            Kind::Table(TableFunction {})
+        }
+        None => {
+            return Err(ErrorCode::InvalidParameterValue(
+                "return type must be specified".to_string(),
+            )
+            .into())
+        }
+    };
+
+    let mut arg_types = vec![];
+    for arg in args.unwrap_or_default() {
+        arg_types.push(bind_data_type(&arg.data_type)?);
+    }
+
+    // resolve database and schema id
+    let session = &handler_args.session;
+    let db_name = session.database();
+    let (schema_name, function_name) = Binder::resolve_schema_qualified_name(db_name, name)?;
+    let (database_id, schema_id) = session.get_database_and_schema_id_for_create(schema_name)?;
+
+    // check if function exists
+    if (session.env().catalog_reader().read_guard())
+        .get_schema_by_id(&database_id, &schema_id)?
+        .get_function_by_name_args(&function_name, &arg_types)
+        .is_some()
+    {
+        let name = format!(
+            "{function_name}({})",
+            arg_types.iter().map(|t| t.to_string()).join(",")
+        );
+        return Err(CatalogError::Duplicated("function", name).into());
+    }
+
+    // Parse function body here
+    // Note that the parsing here is just basic syntax / semantic check, the result will NOT be stored
+    // e.g., The provided function body contains invalid syntax, return type mismatch, ..., etc.
+    let Ok(_ast) = Parser::parse_sql(body.as_str()) else {
+        return Err(ErrorCode::InvalidInputSyntax(
+            "invalid function definition, please recheck your function body in as clause"
+                .to_string(),
+        )
+        .into());
+    };
+
+    // Create the actual function, will be stored in function catalog
+    let function = Function {
+        id: FunctionId::placeholder().0,
+        schema_id,
+        database_id,
+        name: function_name,
+        kind: Some(kind),
+        arg_types: arg_types.into_iter().map(|t| t.into()).collect(),
+        return_type: Some(return_type.into()),
+        language,
+        identifier: body,
```

Comment:
I suggest to use a new field `body: String`, as we may support inline body for Python soon.

Reply:
I see, so the current `identifier` would be specific to general udf. (i.e., external source udf)

Reply:
And I prefer putting this change to a separate PR. (Maybe I'll come up with this tomorrow 🤣

Reply:
Okay!